enginetest/queries/script_queries.go

```diff
@@ -1610,6 +1610,47 @@ var ScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "Ensure scale is not rounded when inserting to DECIMAL type through float64",
+		SetUpScript: []string{
+			"create table test (number decimal(40,16));",
+			"insert into test values ('11981.5923291839784651');",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SELECT COUNT(*) FROM test WHERE number = CONVERT('11981.5923291839784651', DECIMAL)",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:    "INSERT INTO test VALUES (11981.5923291839784651);",
+				Expected: []sql.Row{{sql.NewOkResult(1)}},
+			},
+			{
+				Query:    "SELECT COUNT(*) FROM test WHERE number = CONVERT('11981.5923291839784651', DECIMAL)",
+				Expected: []sql.Row{{2}},
+			},
+			{
+				Query:    "INSERT INTO test VALUES (119815923291839784651.11981592329183978465111981592329183978465144);",
```

Comment:
Nice! Good test for a boundary condition. You should try moving all the allowed digits to the right of the decimal point so that we can really stress test our handling of the decimal portion, and you should add a check to make sure we get the correct expected value back without rounding.